File: WorldFeed/Server/src/WorldFeed.Common/DTO/TweetWithSearchMetadataDTO.cs
namespace WorldFeed.Common.DTO
{
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;

    public class TweetWithSearchMetadataDTO : TweetDTO, ITweetWithSearchMetadataDTO
    {
        [JsonProperty("metadata")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public ITweetFromSearchMetadata TweetFromSearchMetadata { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/TweetWithSearchMetadataDTO.ts
import {TweetDTO} from "./TweetDTO";
import {ITweetFromSearchMetadata} from "../../Public/Models/Interfaces/DTO/ITweetFromSearchMetadata";
import {ITweetWithSearchMetadataDTO} from "../../Public/Models/Interfaces/DTO/ITweetWithSearchMetadataDTO";

export class TweetWithSearchMetadataDTO extends TweetDTO implements ITweetWithSearchMetadataDTO {
  // [JsonProperty("metadata")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public tweetFromSearchMetadata: ITweetFromSearchMetadata;
}
